### PR TITLE
Closes #701 by adding documentation about the Payment RPC interface.

### LINKED ISSUE #701: RPC command documentation
Document the new RPC interface.

Here's the original description for posterity:

> We'd like people to experiment with the RPC to build cool mini wallets, wrappers to make things easier, etc. We should document our existing RPC commands better.
> 
> Also, we should make *new* RPC commands so that users don't need to exchange the encrypted buckets, which would allow more experimentation.
> 
> My proposal:
> 
> * zcrawscan (`zcsecretkey`, `height`=0) scans the blockchain for buckets spendable from this key starting at `height` and returns them all alongside their corresponding serials.
> * zcrawcheck (`serial`) returns whether or not the serial exists in `mapSerials` to show whether or not the bucket can be spent.
>
> (Some thought should be put into what buckets are returned from the mempool?)

### PR DESCRIPTION
**Pull request by bitcartel**
@daira Calls which might not make it into z9 are labelled (NOT IN Z9 ALPHA).

### CODE REVIEW

**Comment by daira** [comment]
Hmm. We should make sure not to raise expectations for the new API in this release, since you can't actually use it to send funds yet, and therefore people will have to continue using the old API until Beta 1.

On an API design issue, there should be idempotent versions of the operation status queries that don't consume the status for completed operations.

**Comment by daira** [comment]
ACK (no coverage check needed). If any of the calls marked "NOT IN Z9 ALPHA" do get in then we'll update this doc.

**Comment by bitcartel** [comment]
Public alpha guide should be updated:

1. llink to the payment doc mentioning that only some of the APIs are available for use
2. use the new calls when appropriate e.g. z_getnewaddress, z_listaddresses instead of having to do a zcrawkeygen and manually keep track of keys.

File: doc/payment-api.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+Zcash supports all commands in the Bitcoin Core API (as of version 0.11.2).   Where applicable, Zcash will extend commands in a backwards-compatible way to enable additional functionality.
+
+We do not recommend use of accounts which are now deprecated in Bitcoin Core.  Where the account parameter exists in the API, please use “” as its value.
+
+To support multiple users in a single node’s wallet, consider using getnewaddress or z_getnewaddress to obtain a new address for each user.  Also consider mapping multiple addresses to each user.
+
+## List of Zcash API commands
+
+Optional parameters are denoted in [square brackets].
```
Comment (defuse) [inline doc/payment-api.md]: Ignore [this is a test]

**Comment by defuse** [comment]
Is it possible to avoid using `<br />` in the Markdown? Is it to achieve some kind of special formatting, or would actual line breaks in the file be equivalent?

**Comment by defuse** [comment]
ACK despite the nitpick above

**Comment by daira** [comment]
ACK https://github.com/zcash/zcash/pull/1292/commits/5f3f741e472cfedf9129829cdedff836f6b10632

**Comment by daira** [comment]
@zkbot r+

**Pull request merged**